api/v1alpha3/common_types.go

```diff
@@ -24,6 +24,10 @@ const (
 	// ClusterLabelName is the label set on machines linked to a cluster and
 	// external objects(bootstrap and infrastructure providers)
 	ClusterLabelName = "cluster.x-k8s.io/cluster-name"
+
+	// PauseReconcilerAnnotation is an annotation that can be applied to any Cluster API
+	// object to pause and skip reconciliation.
```

Comment:
It might be good to say that the controller for the resource must support the annotation as well.